packages/orchestration/test/exos/zcf-for-flows.test.ts

```diff
@@ -0,0 +1,185 @@
+import { E, Far } from '@endo/far';
+import type { TestFn } from 'ava';
+import { createRequire } from 'node:module';
+import { prepareZcfForFlows } from '../../src/exos/zcf-for-flow.js';
+import { provideDurableZone } from '../supports.js';
+
+const nodeRequire = createRequire(import.meta.url);
+const contractEntry = nodeRequire.resolve('../fixtures/zcfTester.contract.js');
+
+const makeTestContext = async () => {
+  let testJig;
+  const setJig = jig => (testJig = jig);
+  const fakeVatAdmin = makeFakeVatAdmin(setJig);
+  const { zoeService: zoe, feeMintAccess } = makeZoeKitForTest(
+    fakeVatAdmin.admin,
+  );
+
+  const bundleCache = await makeNodeBundleCache('bundles', {}, s => import(s));
+  const contractBundle = await bundleCache.load(contractEntry);
+
+  fakeVatAdmin.vatAdminState.installBundle('b1-contract', contractBundle);
+  const installation = await E(zoe).installBundleID('b1-contract');
+
+  const stuff = makeIssuerKit('Stuff');
+  await E(zoe).startInstance(installation, { Stuff: stuff.issuer });
+  assert(testJig, 'startInstance did not call back to setTestJig');
+
+  const zcf: ZCF = testJig.zcf;
+
+  const zone = provideDurableZone('root');
+  const vt = prepareSwingsetVowTools(zone);
+  const zcfLtd = prepareZcfForFlows(zcf, zone, vt);
+  return { zoe, zcf, stuff, feeMintAccess, zcfLtd, vt };
+};
+
+type TestContext = Awaited<ReturnType<typeof makeTestContext>>;
+
+const test = anyTest as TestFn<TestContext>;
+
+test.before('set up context', async t => (t.context = await makeTestContext()));
+
+test('yes as is: atomicRearrange(), ... getTerms()', async t => {
+  const { zcf, zcfLtd } = t.context;
+
+  t.notThrows(() => zcfLtd.atomicRearrange([]));
+
+  t.notThrows(() => zcfLtd.assertUniqueKeyword('K1'));
+  t.throws(() => zcfLtd.assertUniqueKeyword('Stuff'));
+
+  t.deepEqual(zcfLtd.getTerms(), zcf.getTerms());
+});
+
+test.todo('reallocate - yes as is');
+
+test('makeEmptySeatKit: remove userSeat', async t => {
+  const { zcf, zcfLtd } = t.context;
+
+  const kit = zcfLtd.makeEmptySeatKit();
+  t.deepEqual(Object.keys(kit), ['zcfSeat']);
+
+  const { zcfSeat } = kit;
+  t.falsy(zcfSeat.hasExited(), 'zcfSeat works as usual');
+});
+
+test('makeInvitation: watch promise', async t => {
+  const { zoe, zcf, zcfLtd, vt } = t.context;
+
+  const handler = Far('Trade', { handle: seat => {} });
+  const toTradeVow = zcfLtd.makeInvitation(handler, 'trade');
+
+  const toTrade = await vt.when(toTradeVow);
+  const amt = await E(E(zoe).getInvitationIssuer()).getAmountOf(toTrade);
+  t.like(amt, { value: [{ description: 'trade' }] });
+});
+
+test('makeInvitation: non-passable handler', async t => {
+  const { zoe, zcf, zcfLtd, vt } = t.context;
+
+  const toTradeVow = zcfLtd.makeInvitation(_seat => {}, 'trade');
+
+  const toTrade = await vt.when(toTradeVow);
+  const amt = await E(E(zoe).getInvitationIssuer()).getAmountOf(toTrade);
+  t.like(amt, { value: [{ description: 'trade' }] });
+});
```

Comment:
This is unnecessary to test for as it's not explicitly supported. Only passable arguments without promises are supported by the async flow membrane, the attenuated zcf should never see such a thing. Let's actually make the guard stricter than the original zcf and test that it refuses a function handler.